prime-router/src/test/kotlin/fhirengine/translation/hl7/utils/CustomFHIRFunctionsTests.kt

```diff
@@ -662,12 +662,52 @@ class CustomFHIRFunctionsTests {
 
     @Test
     fun `test convertDateToAge 0 days`() {
+        val randomDate = Date(1687464192857)
+        val calendar = Calendar.getInstance()
+        calendar.time = randomDate
+
+        val ageList = CustomFHIRFunctions.convertDateToAge(
+            mutableListOf(DateType(calendar.time)),
+            mutableListOf(mutableListOf(DateType(calendar.time), StringType("day")))
+        )
+
+        val age = ageList[0]
+        assertThat(age is Age).isEqualTo(true)
+        if (age is Age) {
+            assertThat(age.unit).isEqualTo("day")
+            assertThat(age.value.toInt()).isEqualTo(0)
+            assertThat(age.code).isEqualTo("d")
+        }
+    }
+
+    @Test
+    fun `test convertDateToAge 0 days - default`() {
+        val randomDate = Date(1687464192857)
+        val calendar = Calendar.getInstance()
+        calendar.time = randomDate
+
+        val ageList = CustomFHIRFunctions.convertDateToAge(
+            mutableListOf(DateType(randomDate)),
+            mutableListOf(mutableListOf(DateType(randomDate)))
+        )
+
+        val age = ageList[0]
+        assertThat(age is Age).isEqualTo(true)
+        if (age is Age) {
+            assertThat(age.unit).isEqualTo("day")
+            assertThat(age.value.toInt()).isEqualTo(0)
+            assertThat(age.code).isEqualTo("d")
+        }
+    }
```

Comment:
These two tests are super similar, and I think could use some tiny edits:

- The calendar seems unnecessary since `DateType(date)` seems to work
- `randomDate` implies randomness in the test, I'd maybe recommend `someDate` to avoid misunderstandings
- Since the result should be the same for each case, it might make sense to combine these into one test

Took a stab at how that might look:

```suggestion
    @Test
    fun `test convertDateToAge 0 days`() {
        val someDate = Date(1687464192857)

        val results = listOf(
            // Explicitly look for "day"
            CustomFHIRFunctions.convertDateToAge(
                mutableListOf(DateType(someDate)),
                mutableListOf(mutableListOf(DateType(someDate), StringType("day")))
            ),
            // Get default response without specifying type
            CustomFHIRFunctions.convertDateToAge(
                mutableListOf(DateType(someDate)),
                mutableListOf(mutableListOf(DateType(someDate)))
            )
        )

        results.forEach { ageList ->
            val age = ageList[0]
            assertThat(age is Age).isEqualTo(true)
            if (age is Age) {
                assertThat(age.unit).isEqualTo("day")
                assertThat(age.value.toInt()).isEqualTo(0)
                assertThat(age.code).isEqualTo("d")
            }
        }
    }
```